services/provider/server/storageclusterpeer.go

```diff
@@ -0,0 +1,39 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type storageClusterPeerManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newStorageClusterPeerManager(cl client.Client, namespace string) (*storageClusterPeerManager, error) {
+	return &storageClusterPeerManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (s *storageClusterPeerManager) GetByPeerStorageClusterUID(ctx context.Context, storageClusterUID types.UID) (*ocsv1.StorageClusterPeer, error) {
+	storageClusterPeerList := &ocsv1.StorageClusterPeerList{}
+	err := s.client.List(ctx, storageClusterPeerList, client.InNamespace(s.namespace))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list StorageClusterPeer: %v", err)
+	}
+	for i := range storageClusterPeerList.Items {
+		storageClusterPeer := &storageClusterPeerList.Items[i]
+		if storageClusterPeer.Status.PeerInfo != nil {
+			if storageClusterUID == types.UID(storageClusterPeer.Status.PeerInfo.StorageClusterUid) {
+				return storageClusterPeer, nil
+			}
+		}
```

Comment:
No need for all of the nesting nesting

```suggestion
		peerInfo :=  storageClusterPeerList.Items[i].Status.PeerInfo
		if peerInfo != nil &&  types.UID(peerInfo.storageClusterUID) == storageClusterUID {
			return storageClusterPeer, nil
		}
```